engine/renderer/private/particles/particle_pipeline.cpp

```diff
@@ -291,7 +396,12 @@ void ParticlePipeline::CreatePipelines()
         pipelineLayoutCreateInfo.setLayoutCount = descriptorLayouts.size();
         pipelineLayoutCreateInfo.pSetLayouts = descriptorLayouts.data();
 
-        pipelineLayoutCreateInfo.pushConstantRangeCount = 0;
+        vk::PushConstantRange pcRange {};
+        pcRange.stageFlags = vk::ShaderStageFlagBits::eVertex;
```

Comment:
You should use the PipelineBuilder class for generating this graphics pipeline, it makes it much shorter and removes a lot of bloat. It also automatically infers vertex structure from the spv binaries. The builder will also constructor descriptor layouts for you, but this can be annoying, so after you make your descriptor set layout you can cache it using a static function on the pipeline builder.